docs/src/tutorial.rst

```diff
@@ -0,0 +1,487 @@
+
+This creates *my_widget_service.ts* in the *bin* directory.
+We don't need most of this code,
+so for now change it to the following:
+
+.. code-block:: ts
+
+    #!/usr/bin/env node
+    import cdk = require('@aws-cdk/cdk');
+
+    class MyWidgetServiceStack extends cdk.Stack {
+      constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
+        super(parent, name, props);
+
+
+      }
+    }
+
+    // Create a new CDK app
+    const app = new cdk.App();
+
+    // Add your stack to it
+    new MyWidgetServiceStack(app, 'MyWidgetServiceStack');
+
+    app.run();
+
+Save it and make sure it builds and creates an empty stack.
+
+.. code-block:: sh
+
+    npm run build
+    cdk synth
+
+You should see a stack like the following,
+where CDK-VERSION is the version of the CDK.
+
+.. code-block:: sh
+
+    Resources:
+      CDKMetadata:
+        Type: 'AWS::CDK::Metadata'
+        Properties:
+          Modules: >-
+            @aws-cdk/cdk=CDK-VERSION,@aws-cdk/cx-api=CDK-VERSION,my_widget_service=0.1.0
+
+.. _create_lambda_functions:
+
+Step 2: Create a |LAM| Function to List all Widgets
+===================================================
+
+The next step is to create a |LAM| function to list all of the widgets in our
+|S3| bucket.
+
+Create the directory *resources* at the same level as the *bin* directory.
+
+.. code-block:: sh
+
+    mkdir resources
+
+Create the following Javascript file, *widgets.js*,
+in the *resources* directory.
+
+.. code-block:: js
+
+    const AWS = require('aws-sdk');
+    const S3 = new AWS.S3();
+
+    const bucketName = process.env.BUCKET;
+
+    // For now we only handle GET /:
+    exports.main = async function(event, context) {
+      try {
+        var method = event.httpMethod;
+
+        if (method == "GET") {
+          if (event.path.length == 1) {
+            const data = await S3.listObjectsV2({ Bucket: bucketName }).promise();
+            var body = {
+              widgets: data.Contents.map(function(e) { return e.Key })
+            };
+            return {
+              statusCode: 200,
+              headers: {},
+              body: JSON.stringify(body)
+            };
+          }
+          else {
+            // GET /name
+            var name = event.path.substring(1, event.path.length);
+  
+            const data = await S3.getObject({ Bucket: bucketName, Key: name}).promise();
+            var body = data.Body.toString('utf-8');
+
+            return {
+              statusCode: 200,
+              headers: {},
+              body: JSON.stringify(body)
+            };
+          }
+        }
+        else {
+          var body = "We only accept GET, POST, and DELETE, not " + method;
+        
+          return {
+            statusCode: 200,
+            headers: {},
+            body: JSON.stringify(body)
+          };
+        }
+      } catch(error) {
+        var body = error.stack || JSON.stringify(error, null, 2);
+        var response = makeError(body);
```

Comment:
Again, where are you showing me what code to copy/paste to obtain this `makeError` function?

The code will not be complete without this.

Reply:
That ties into another issue that I raised: how will people know it's an error response?

Either set `statusCode: 500` or make body a JSON object. 

If you do make body a JSON object, I suspect you also need to set a `content-type` header.